packages/contracts-sdk/src/lib/addresses.ts

```diff
@@ -22,6 +22,164 @@ export interface TokenInfo {
   isNewPKP: boolean;
 }
 
+/**
+ * Converts a public key between compressed and uncompressed formats.
+ *
+ * @param publicKey - Public key as a Buffer (33 bytes compressed or 65 bytes uncompressed)
+ * @param compressed - Boolean flag indicating whether the output should be compressed
+ * @returns Converted public key as a Buffer
+ */
+export function publicKeyConvert(
+  publicKey: Buffer,
+  compressed: boolean = true
+): Buffer {
+  if (compressed) {
+    // Compress the public key (if it's not already compressed)
+    if (publicKey.length === 65 && publicKey[0] === 0x04) {
+      const x = publicKey.subarray(1, 33);
+      const y = publicKey.subarray(33, 65);
+      const prefix = y[y.length - 1] % 2 === 0 ? 0x02 : 0x03;
+      return Buffer.concat([Buffer.from([prefix]), x]);
```

Comment:
nit: Ideally we move away from `Buffer` to `TextEncoder` as `Buffer` is a node utility. Can we refactor this if possible?